### PR TITLE
Migrate the article about public access in vpc

### LINKED ISSUE #394: Public access for Kafka in a VPC
# What should we add to our content collections?

Migrate https://help.aiven.io/en/articles/4332080-public-access-for-kafka-in-a-vpc

# Is this something you would work on yourself?

[x] I plan to open a pull request for this

### PR DESCRIPTION
**Pull request by anelook**
Fixes #394

### CODE REVIEW

File: docs/products/kafka/howto/public-access-in-vpc.rst [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+To enable public access for **Aiven for Apache Kafka** service which is running within a virtual private cloud (VPC), follow these steps:
+
+#. Log in to the Aiven web console and select your **Aiven for Apache Kafka** service.
+#. On the *Overview* page, scroll down to the *Advanced configuration* section and click **Add configuration option** .
+#. Select ``public_access.kafka`` and switch it on. For **Aiven for Apache Kafka Connect** or **Aiven for Apache Kafka Rest**, select ``public_access.kafka_connect`` or ``public_access.kafka_rest`` respectively.
+#. Click **Save advanced configuration**. The *Overview* page now has an **Access Route** setting inside the *Connection information* with **Public** and **Dynamic** options.
+#. Select **Public** to see the public URL for your service.
+
+The connection with the **Dynamic** option is not possible outside the VPC, while the connection with the **Public** option is accessible over the public internet. **Public** connections are still obliged to the service's IP Allow List.
```
Comment (encima) [inline docs/products/kafka/howto/public-access-in-vpc.rst:12]: Change `obliged` to something like "The IP Allow list applies to all connection types (Dynamic and Public, in this example)"

File: docs/products/kafka/howto/public-access-in-vpc.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Enable public access in a VPC
+==============================
+
+To enable public access for **Aiven for Apache Kafka** service which is running within a virtual private cloud (VPC), follow these steps:
```
Comment (encima) [inline docs/products/kafka/howto/public-access-in-vpc.rst:4]: Public Access is not only for Kafka, could this be a generic example (or specify this example uses Kafka but we support it in other services, as well as service components)

File: docs/products/kafka/howto/public-access-in-vpc.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+#. Log in to the Aiven web console and select your **Aiven for Apache Kafka** service.
+#. On the *Overview* page, scroll down to the *Advanced configuration* section and click **Add configuration option** .
+#. Select ``public_access.kafka`` and switch it on. For **Aiven for Apache Kafka Connect** or **Aiven for Apache Kafka Rest**, select ``public_access.kafka_connect`` or ``public_access.kafka_rest`` respectively.
+#. Click **Save advanced configuration**. The *Overview* page now has an **Access Route** setting inside the *Connection information* with **Public** and **Dynamic** options.
```
Comment (encima) [inline docs/products/kafka/howto/public-access-in-vpc.rst:9]: Screenshot? Or is that more trouble because keeping it up to date is additional work?

**Review by encima — CHANGES_REQUESTED** [review]

File: docs/products/kafka/howto/public-access-in-vpc.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
+Enable public access in a VPC
+==============================
+
+To enable public access for **Aiven for Apache Kafka** service which is running within a virtual private cloud (VPC), follow these steps:
+
+#. Log in to the Aiven web console and select your **Aiven for Apache Kafka** service.
+#. On the *Overview* page, scroll down to the *Advanced configuration* section and click **Add configuration option** .
+#. Select ``public_access.kafka`` and switch it on. For **Aiven for Apache Kafka Connect** or **Aiven for Apache Kafka Rest**, select ``public_access.kafka_connect`` or ``public_access.kafka_rest`` respectively.
```
Comment (dbeech) [inline docs/products/kafka/howto/public-access-in-vpc.rst:8]: "Aiven for Apache Kafka Rest" is not a product name I recognise. Maybe replace with "Karapace REST API" or "Karapace REST Proxy" ?

File: docs/products/kafka/howto/public-access-in-vpc.rst (hunk 3 above)
Comment (anelook) [reply docs/products/kafka/howto/public-access-in-vpc.rst:9]: Yes, and we don't do it for other cases at the moment. So I left this one without the screenshot as well.

File: docs/products/kafka/howto/public-access-in-vpc.rst (hunk 2 above)
Comment (anelook) [reply docs/products/kafka/howto/public-access-in-vpc.rst:4]: Great point. Refactored to make generic

File: docs/platform/howto/public-access-in-vpc.rst [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Enable public access in a VPC
+==============================
+
+To enable public access for a service which is running within a virtual private cloud (VPC), follow these steps:
+
+#. Log in to the Aiven web console and select your service.
+#. On the *Overview* page, scroll down to the *Advanced configuration* section and click **Add configuration option** .
```
Comment (laysauchoa) [inline docs/platform/howto/public-access-in-vpc.rst:7]:
```suggestion
#. On the *Overview* page, scroll down to the *Advanced configuration* section and click **Add configuration option**.
```

File: docs/platform/howto/public-access-in-vpc.rst [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+#. Select an option which starts with ``public_access.`` and follows with the type of service you use and switch it on.
```
Comment (laysauchoa) [inline docs/platform/howto/public-access-in-vpc.rst:8]:
```suggestion
#. Select an option that starts with ``public_access`` followed by the type of service you use and switch it on.
```

File: docs/platform/howto/public-access-in-vpc.rst [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+#. Click **Save advanced configuration**. The *Overview* page now has an **Access Route** setting inside the *Connection information* with **Public** and **Dynamic** options.
+#. Select **Public** to see the public URL for your service.
+
+The connection with the **Dynamic** option is not possible outside the VPC, while the connection with the **Public** option is accessible over the public internet. *The IP allow list* applies to all connection types (Dynamic and Public, in this example).
+
+.. note:: You can change the **public_access** settings without any downtime for your service.
```
Comment (laysauchoa) [inline docs/platform/howto/public-access-in-vpc.rst:14]:
```suggestion
.. note:: You can change the **public_access** settings without any service downtime.
```

**Comment by laysauchoa** [comment]
Hey, what kind of help you need here?

**Comment by anelook** [comment]
Added improvements to the content, @encima, could you have another look?

**Review by encima — APPROVED** [review]

**Pull request merged**